### ISSUE TITLE
Version policies

### ISSUE THREAD

**Issue by rgraff** [ash-project/ash_paper_trail#13]
A couple issues:

* Version resources are created with an `authorize?: false` but does this fail if the api has `authorize :always`
* Are policies authorized currently or are they globably readable (footgun)
* Policies can be added via a mixin. Should update the readme.
* Can we declare policies in the paper_trail section. Might be a lift not worth doing if we can get full support via mixin.

**Comment by zachdaniel** [comment]
Yeah, I think mixins are the way to go for version policies. You're right that it could be a footman that there are no policies on that resource by default, but I'm not sure yet how to handle that ergonomically. Yes, `authorize :always` will cause version resources not to fail, but what we can do is set some context i.e `%{ash_paper_trail_action?: true}` and have a policy that checks for that, so users with `authorize :always` have a way to approve the actions done by this extension.

**Comment by rgraff** [comment]
Based on that input I think we should:

- [ ] Add authorizer and bypass policy for a context (%{ash_paper_trail_action?: true}) to essentially make them create only
- [ ] Add a resource option to make them always readable, scoped only by tenant (if applicable)
- [ ] Update readme with example of how to apply policies via mixin for more fine-grained read controls.

**Comment by frankdugan3** [comment]
I'm needing to specify policies for versions in my app, so I'm implementing a PR. Since this issue is so old, I'm checking in to make sure the proposal is still sound. Here's my thinking:

- The default should basically be to effectively have the bypass policy to be create-only (as mentioned above). This could be gated behind a future-compatibility config option, following Ash's general convention for similar breaking changes.
- My most common use case will be that I want the versions to have the same read policies as the source, so it would be ideal to have a simple DSL option to enable that case.
- Documented mixin support for more complex policies

```elixir
paper_trail do
  # The default could be changed via application config
  version_policies :none (default) | :create_only | :source_read_policy
end
```

Thoughts?

**Comment by zachdaniel** [comment]
`:source_read_policy` would be pretty hard to implement wouldn't it? I'm not opposed but it sounds fairly complex.

**Comment by aVikingTrex** [comment]
I found this issue looking around online for policies on versions. I have seen e16cbcca6fe57c3c686ef5ec42b3c8d47a747111 linked from @frankdugan3. After looking at the bypass solution there, and considering the footgun of policy-less versions by default exists, I am thinking that maybe by default versions can just have `can_read(:parent)` policy.

**Comment by zachdaniel** [comment]
Yeah, we just added `can_read/1`. We can't make it a default without a major version release of the package, but yes that would probably be the best way 👍

**Comment by zachdaniel** [comment]
We can start off by making it opt-in and can go from there?

**Comment by frankdugan3** [comment]
FWIW, I never made a PR because I ended up using a custom extension due to a number of other requirements so I didn't get to test drive it adequately.

**Comment by aVikingTrex** [comment]
Yeah I realised after I suggested can_read how new it was comparitive to this whole thread. I will make a PR today for read policy. 
Though I just noticed too that versions are generated with update actions. Given my conceptual model for this is audit trails, why do we support updates as it seems antithetical? Should we consider policies around updates this coming PR or a separate PR?

**Comment by zachdaniel** [comment]
🤔 Yeah that is fairly strange. Can you find anywhere internally we use said update action? We probably should not generate this.

**Comment by aVikingTrex** [comment]
No the tests I have added in the PR reference it only to assert we dont change behavior. The only other reference is the version transformer creating the update action itself. From what I see the only operation we actually do is only [create/bulk_create](https://github.com/ash-project/ash_paper_trail/blob/d46dd247dca649f350f6d34cd6f1a80886b0335c/lib/resource/changes/create_new_version.ex#L344).

Do you think just drop the tests around update, and the action definition in the transformer?
